dlrover/python/master/node/worker.py

```diff
@@ -305,3 +311,113 @@ def verify_restarting_training(self, node_id):
             # Set False to avoid restart repeatedly.
             worker.restart_training = False
         return restart
+
+    def is_training_hang_by_pending(self) -> bool:
+        """
+        To prevent training hanging by pending. Should exit when there is
+        inextricable pending issue.
+
+        There is 3 main conditions:
+        1: exist pending
+        2: alive nodes number consistently lower than the min nodes requires
+        3: 1+2 last for a certain time
+
+        Return:
+            bool
+        """
+
+        if not self.has_node_required_info():
+            return False
+
+        # pending time as timeout for now
+        timeout = _dlrover_context.seconds_to_wait_pending_pod
+        cur_nodes = list(self._nodes.values())
+
+        # collect pending and running nodes
+        pending_nodes: List[Node] = []
+        running_nodes: List[Node] = []
+        for node in cur_nodes:
+            if node is None or node.is_released or node.create_time is None:
+                continue
+
+            if node.status == NodeStatus.PENDING:
+                pending_nodes.append(node)
+            elif node.status == NodeStatus.RUNNING:
+                running_nodes.append(node)
+
+        # with condition 1 + 2
+        if (
+            len(pending_nodes) == 0
+            or len(running_nodes) >= self._nodes_required[0]
```

Comment:
`len(running_nodes) >= self._min_nodes`. `self._nodes_required[0]` is not readable.

Reply:
use function to express self._nodes_required[0] instead